IdentityCore/src/logger/MSIDLogger+Trace.h

```diff
@@ -0,0 +1,28 @@
+//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+#define MSID_TRACE // Unused
```

Comment:
can you define it in the main logger file?

Reply:
I had it in MSIDLogger+Internal.h but to override the trace macro for CPP I also needed to duplicate all of the other macro definitions in this header as well, which we're trying to avoid. It's a lot cleaner to override with it being in it's own header. This lets us use the common MSIDLogger+Internal.h unmodified.

Reply:
Can it go to a function in MSIDLogger+Internal.h that's just not implemented in OBJ-C?

Reply:
I believe I see a way to do it, though I would still also need to use a macro to expand __SHORTFILE__ and __func__ correctly. I'll check on this.

Reply:
Updated